File: fm-domain/src/test/java/io/zeitwert/fm/test/model/enums/CodeTestType.java
package io.zeitwert.fm.test.model.enums;

import io.dddrive.core.enums.model.Enumeration;
import io.dddrive.core.enums.model.base.EnumeratedBase;

/**
 * CodeTestType enum using the NEW dddrive framework.
 * Used for testing enum properties without disturbing other enums like CodeCountry.
 */
public final class CodeTestType extends EnumeratedBase {

    public CodeTestType(Enumeration<CodeTestType> enumeration, String id, String name) {
        super(enumeration, id, name);
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (other == null || getClass() != other.getClass()) return false;
        CodeTestType that = (CodeTestType) other;
        return getId().equals(that.getId());
    }

    @Override
    public int hashCode() {
        return getId().hashCode();
    }
}

File: fm-domain/src/test/java/io/zeitwert/fm/test/model/enums/CodeTestType.kt
package io.zeitwert.fm.test.model.enums

import io.dddrive.enums.model.EnumeratedEnum
import io.dddrive.enums.model.base.EnumerationBase

enum class CodeTestType(
	override val defaultName: String,
) : EnumeratedEnum {

	TYPE_A("Type A"),
	TYPE_B("Type B"),
	TYPE_C("Type C"),
	;

	override val enumeration get() = Enumeration

	companion object Enumeration : EnumerationBase<CodeTestType>(CodeTestType::class.java) {
		init {
			entries.forEach { addItem(it) }
		}

		fun getTestType(itemId: String?) = if (itemId != null) getItem(itemId) else null
	}
}
